File: 3884-minimum-absolute-difference-in-sliding-submatrix/minimum-absolute-difference-in-sliding-submatrix.py
class Solution:
    def minAbsDiff(self, grid: List[List[int]], k: int) -> List[List[int]]:
        m,n= len(grid),len(grid[0])
        ans = [[0]*(n-k+1) for t in range(m-k+1)]
        if k == 1:
            return ans
        for i in range(m-k+1):
            for j in range(n-k+1):
                temp = []
                for x in range(i,i+k):
                    for y in range(j,j+k):
                        temp.append(grid[x][y])

                temp = list(set(temp))
                temp.sort()
                if len(temp) == 1:
                     ans[i][j] =0
                else:
                    d = 1e10
                    for z in range(len(temp)-1):
                        d = min(d,abs(temp[z]-temp[z+1]))
                    ans[i][j] = d 
        return ans 

File: 3884-minimum-absolute-difference-in-sliding-submatrix/minimum-absolute-difference-in-sliding-submatrix.cpp
class Solution {
public:
    vector<vector<int>> minAbsDiff(vector<vector<int>>& grid, int k) {
        int m= grid.size() ;  int n = grid[0].size() ; 
         vector<vector<int>> ans(m - k + 1, vector<int>(n - k + 1,0));
         if(k == 1) return ans ;
         for(int i=0;i<=m-k;i++){ vector<int> res ; 
            for(int j=0;j<=n-k;j++){  vector<int> temp ;
                for(int x=i;x<i+k;x++){
                    for(int y=j;y<j+k;y++){
                        temp.push_back(grid[x][y]) ;
                    }
                } int d =1e6; sort(temp.begin(),temp.end()) ;
                 temp.erase(unique(temp.begin(),temp.end()),temp.end()) ;
                for(int z = 0;z<temp.size()-1;z++) d = min(d,abs(temp[z]-temp[z+1])) ;
                if(temp.size() == 1) d = 0 ;
                ans[i][j] = d ;
         } 
       }  return ans ;
    }
};